opensrp-core/src/test/java/org/smartregister/service/HTTPAgentTest.java

```diff
@@ -1221,4 +1222,14 @@ public void testHttpImagePostConfiguresConnectionRequestCorrectly() throws Excep
 
         Mockito.verify(printWriter, Mockito.times(7)).flush();
     }
+
+    @Test
+    public void testOauth2authenticateEncodesPasswordCorrectly() {
+        HTTPAgent httpAgentSpy = Mockito.spy(httpAgent);
+
+        StringBuffer stringBuffer = ReflectionHelpers.callInstanceMethod(httpAgentSpy, "getRequestParams",
+                ReflectionHelpers.ClassParameter.from(String.class, "testUser"),
+                ReflectionHelpers.ClassParameter.from(char[].class, "abc123%^&.".toCharArray()));
+        Assert.assertEquals("&username=testUser&password=abc123%25%5E%26.", stringBuffer.toString());
```

Comment:
I thought the password field should not be encoded. The change was to make sure the only field that is encoded is username and the password field remains as entered

Reply:
Password needs to be encoded so that special characters in the password string get escaped